src/streaming/net/XHRLoader.js

```diff
@@ -45,53 +46,56 @@ function XHRLoader(cfg) {
 
     function load(httpRequest) {
 
-        // Variables will be used in the callback functions
-        const requestStartTime = new Date();
-        const request = httpRequest.request;
+        modifyRequest(httpRequest, requestModifier)
+            .then(() => {
+                // Variables will be used in the callback functions
+                const requestStartTime = new Date();
```

Comment:
Same as fetch loader, should the requestStartTime include asynchronous modifications?